core.js

```diff
@@ -2761,7 +2761,190 @@ HTMLAudioElement.HAVE_METADATA = HTMLMediaElement.HAVE_METADATA;
 HTMLAudioElement.HAVE_CURRENT_DATA = HTMLMediaElement.HAVE_CURRENT_DATA;
 HTMLAudioElement.HAVE_FUTURE_DATA = HTMLMediaElement.HAVE_FUTURE_DATA;
 HTMLAudioElement.HAVE_ENOUGH_DATA = HTMLMediaElement.HAVE_ENOUGH_DATA;
+class Video {
+  static getDevices() {
+    return [];
+  }
+}
+
+class MediaDeviceKind {
+}
+MediaDeviceKind.audioInput = "audioinput";
+MediaDeviceKind.audioOutput = "audiooutput";
+MediaDeviceKind.videoInput = "videoinput";
+
+class MediaDeviceInfo {
+  constructor(device, kind) {
+    if (kind === MediaDeviceKind.videoInput) {
+      this.kind = kind;
+      this.deviceId = device.id;
+      this.groupId = device.id;
+      this.label = device.name;
+    } else {
+      throw new Error("not implemented");
+    }
+  }
+}
+
+class MediaStreamTrack {
```

Comment:
So there is [already a `MediaStreamTrack`](https://github.com/webmixedreality/exokit/blob/master/deps/exokit-bindings/webaudiocontext/src/MediaStreamTrack.cpp#L5) in the C++. How does that reconcile with this design where it's implemented in JS as well?

Reply:
It's probably better to implement as much of the spec as possible in JS.

Reply:
Either way, it doesn't seem right to have both implementations. What's the design plan there?

Reply:
I think it makes the most sense to implement MediaStreamTrack in JS, and then expose the low-level APIs via nativeBindings. It's better to do it that way because the selection algorithm is quite complex and better suited to JS. And once you've selected the various characteristics the user wants to use, it's straightforward to pass it along to the native class.

Reply:
I can get behind that, but in that case we shouldn't have the MediaStreamTrack class doubly implemented once in JS here and once in the audio subsystem.

Reply:
That is, one of these should probably be deleted and merged into the other.